lambda/main.go

```diff
@@ -127,3 +147,59 @@ func Handler(ctx context.Context, evt json.RawMessage) (string, error) {
 
 	return "", nil
 }
+
+func initTokenProvider(awsRegion string) (token.Provider, error) {
+	mutuallyExclusiveEnvVars := []string{
+		BKAgentTokenEnvVar,
+		BKAgentTokenSSMKeyEnvVar,
+		BKAgentTokenSecretsManagerSecretIDEnvVar,
+	}
+
+	if err := checkMutuallyExclusiveEnvVars(mutuallyExclusiveEnvVars...); err != nil {
+		return nil, err
+	}
```

Comment:
This is maybe the only non-backwards compatible part of the code. This is because, if someone is using previous versions and setting both `BUILDKITE_AGENT_TOKEN ` and `BUILDKITE_AGENT_TOKEN_SSM_KEY ` they will now get an error whereas before `SSM` will take precedence.

Reply:
That sounds to me like fixing a bug! :)